packages/runtime/runtime-utils/src/summarizerNode/summarizerNode.ts

```diff
@@ -61,10 +62,40 @@ export class SummarizerNode implements IRootSummarizerNode {
 
 	protected readonly children = new Map<string, SummarizerNode>();
 	protected readonly pendingSummaries = new Map<string, SummaryNode>();
-	private wipReferenceSequenceNumber: number | undefined;
+	protected wipReferenceSequenceNumber: number | undefined;
 	private wipLocalPaths: { localPath: EscapedPath; additionalPath?: EscapedPath } | undefined;
 	private wipSkipRecursion = false;
 
+	protected readonly logger: ITelemetryLogger;
+
+	/**
+	 * Do not call constructor directly.
+	 * Use createRootSummarizerNode to create root node, or createChild to create child nodes.
+	 */
+	public constructor(
+		baseLogger: ITelemetryLogger,
+		private readonly summarizeInternalFn: SummarizeInternalFn,
+		config: ISummarizerNodeConfig,
+		private _changeSequenceNumber: number,
+		/** Undefined means created without summary */
+		private _latestSummary?: SummaryNode,
+		private readonly initialSummary?: IInitialSummary,
+		protected wipSummaryLogger?: ITelemetryLogger,
+		/** A unique id of this node to be logged when sending telemetry. */
+		protected telemetryId?: string,
```

Comment:
Moved the constuctor to the top because I can never find it. The only change here is addition of `telemetryId` and the child logger creation in the constructor.

Reply:
should it be called nodeId? or something else that more clearly indicates what this ID is about?

Reply:
Renamed it to "telemetryNodeId". I added the prefix telemetry because it should not be used for anything other than logging.